### PR TITLE
Issue/3970 implement blog settings

### PR DESCRIPTION
**Pull request by SergioEstevao**
#3970 

First stab at implementing Blog settings on the app.

This PR just implements Site Title and Site Tagline.

How To Test:

* Open Blog Details
* Open Settings
* Change the Site Title or Tagline
* Press Save
* See if the change is reflected on the server

This should be tested on Sites hosted on WP.com and Self hosted sites with and without Jetpack.

Needs Review: @koke

### CODE REVIEW

File: MIGRATIONS.md [hunk 1]
```diff
@@ -18,3 +18,9 @@ Migration details:
 
 - Only `Account` objects where `isWpcom == YES` will be migrated, added a predicate filter to the mapping model
 - `Blog` has a custom migration policy to calculate `isHostedAtWPcom` and `username`
+
+## WordPress 35 (@sergioestevao 2015-07-08)
```
Comment (koke) [inline MIGRATIONS.md]: I think it's a common practice to do changelogs with the newer versions at the top. Also, and I know it's most likely not your fault, where is 34?

Comment (SergioEstevao) [inline MIGRATIONS.md]: 34 was  missing but I can add some of the changes that I made on that.

Comment (koke) [inline MIGRATIONS.md]: I just noticed that 35 wasn't added but modified. What else changed from 34 to 35?

**Comment by koke** [comment]
Some issues (tested with a self-hosted/XML-RPC):
- When you edit the title/tagline, you can go back and nothing lets you know that your edits will be lost.
- When you're saving the changes, you can still tap the 'Save' button.
- When you edit the title, save, and go back, the blog details screen still shows the old title.
- The automatic refresh when loading the settings screen feels really weird, I think the refresh shouldn't move the UI.
 - When I open settings, and tap on the title to edit, when the refresh finishes it dismisses the keyboard
 - Since the UI moves when it finishes refreshing, it's likely that I'll tap a different field than the one I was aiming for.
- When I pull to refresh manually, it gets stuck refreshing.

File: WordPress/Classes/Networking/BlogServiceRemoteXMLRPC.m [hunk 2]
```diff
@@ -125,4 +181,15 @@ - (WPXMLRPCRequestOperation *)operationForPostFormatsWithBlog:(Blog *)blog
     return operation;
 }
 
+- (RemoteBlogSettings *)remoteBlogSettingFromXMLRPCDictionary:(NSDictionary *)json
+{
+    RemoteBlogSettings *remoteSettings = [[RemoteBlogSettings alloc] init];
+    
+    remoteSettings.name = [json stringForKeyPath:@"blog_title.value"];
+    remoteSettings.desc = [json stringForKeyPath:@"blog_tagline.value"];
+    //remoteSettings.blogPublic = [json numberForKey:@"blog_public"];
```
Comment (koke) [inline WordPress/Classes/Networking/BlogServiceRemoteXMLRPC.m]: Why is this commented out? What about the other settings (e.g. language)?

I think those belong in a separate PR to begin with, but if we're adding code for them, I think it should be complete.

File: WordPress/Classes/ViewRelated/Blog/EditSiteViewController.m [hunk 3]
```diff
@@ -609,42 +708,34 @@ - (void)save:(id)sender
...
+//    } else {
+//        if ([self.startingUser isEqualToString:self.usernameTextField.text] &&
+//            [self.startingPwd isEqualToString:self.passwordTextField.text] &&
+//            [self.startingUrl isEqualToString:self.urlTextField.text]) {
+//            // No need to check if nothing changed
+//            [self cancel:nil];
+//        } else {
+//            [self validateUrl];
+//        }
+//    }
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Blog/EditSiteViewController.m]: Don't commit commented out code

https://github.com/wordpress-mobile/WordPress-iOS/wiki/WordPress-for-iOS-Objective-C-Style-Guide#before-you-commit

File: WordPress/Classes/Networking/BlogServiceRemoteXMLRPC.m (hunk 2 above)
Comment (SergioEstevao) [inline WordPress/Classes/Networking/BlogServiceRemoteXMLRPC.m]: This PR should be only for site title and site tagline. I will remove those.

**Comment by SergioEstevao** [comment]
@koke ready for the next review round.

**Comment by koke** [comment]
> Tagline isn't synced when a site is added, so it's blank on the first load until it refreshes

This is still happening. I checked the code and I'm confused about the new settings methods:
- On XML-RPC, `syncSettingsForBlog` calls the same method than `syncOptionsForBlog` (`wp.getOptions`)
- On REST, it calls `sites/$site/settings` instead of `sites/$site`, although the results of both look pretty similar.

Could we merge syncOptions and syncSettings?

File: WordPress/Classes/Services/BlogService.m [hunk 4]
```diff
@@ -191,6 +192,38 @@ - (void)syncOptionsForBlog:(Blog *)blog
...
 
+- (void)syncSettingsForBlog:(Blog *)blog
+                    success:(void (^)())success
+                    failure:(void (^)(NSError *error))failure
+{
+    id<BlogServiceRemote> remote = [self remoteForBlog:blog];
+    [remote syncSettingsForBlog:blog
+                        success:^(RemoteBlogSettings *settings) {
+                            blog.blogName = settings.name;
+                            blog.blogTagline = settings.desc;
```
Comment (koke) [inline WordPress/Classes/Services/BlogService.m]: This should get the blog in context with the `objectID` instead of accessing it directly

**Comment by koke** [comment]
The tagline placeholder text doesn't fit in the text field:

screen shot 2015-07-10 at 11 35 52

**Comment by SergioEstevao** [comment]
Regarding the settings call vs options:

 * on the REST APU there is a new call settings that have a lot of values that blog options don’t provide
 * on XML-RPC we only have options

in order to maintain the remote interface the same for both APIs I decided to have a new method call syncSettings, that on REST calls the new settings endpoint and on XML-RPC uses options.

**Comment by koke** [comment]
When you save a new title, the Back button still shows the old title, even after going back to the blog details and opening settings again

ios simulator screen shot 10 jul 2015 11 40 42

**Comment by koke** [comment]
> on the REST API there is a new call settings that have a lot of values that blog options don’t provide

Then, can we drop syncOptions and use syncSettings everywhere?

**Comment by SergioEstevao** [comment]
Unfortunately syncSettings is not a superset of syncOptions, maybe something we need to purpose to the API team?

**Comment by koke** [comment]
Definitely, the REST endpoints could be more consistent. In any case, syncOptions includes the tagline both for REST  and XML-RPC, so we should store that when we get it.

**Comment by SergioEstevao** [comment]
@koke ready for another round.

File: WordPress/Classes/ViewRelated/Blog/EditSiteViewController.m [hunk 5]
```diff
@@ -604,46 +706,33 @@ - (void)validateUrl
...
+    self.blog.blogName = self.siteTitleTextField.text;
+    self.blog.blogTagline = self.siteTaglineTextField.text;
+    self.blog.geolocationEnabled = self.geolocationEnabled;
+    if ([self.blog hasChanges]) {
+        sender.enabled = NO;
+        [SVProgressHUD show];
+        [blogService updateSettingForBlog:self.blog success:^{
+            [SVProgressHUD showInfoWithStatus:NSLocalizedString(@"Settings Saved", @"Message to show when settings are saved")];
+            self.navigationController.navigationBar.backItem.title = self.blog.blogName;
+            self.navigationController.navigationBar.backItem.backBarButtonItem.title = self.blog.blogName;
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Blog/EditSiteViewController.m]: This doesn't seem right. I think the settings controller shouldn't be messing with the navigation stack, instead we should have the blog details controller listen for changes: 

https://gist.github.com/koke/90b814fedf0d92287347

**Comment by koke** [comment]
There was still the issue with being able to go back without saving, but since there's still ongoing talk about the best UI design for that, :+1: for now
